x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java

```diff
@@ -710,4 +673,166 @@ public String getFeatureName() {
     public String getFeatureDescription() {
         return "Manages Watch definitions and state";
     }
+
+    private Settings getWatchesIndexSettings() {
+        return Settings.builder()
+            .put("index.number_of_shards", 1)
+            .put("index.number_of_replicas", 0)
+            .put("index.auto_expand_replicas", "0-1")
+            .put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), 6)
+            .put(IndexMetadata.SETTING_PRIORITY, 800)
+            .build();
+    }
+
+    private XContentBuilder getWatchesIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+
+            builder.startObject();
+            {
+                builder.startObject(SINGLE_MAPPING_NAME);
+                builder.field("dynamic", "strict");
+                {
+                    builder.startObject("_meta");
+                    builder.field("version", Version.CURRENT);
```

Comment:
The current template doesn't have a `version` field under they `_meta` object.
If I read the code correctly in `SystemIndexManager` this means that irregardless of that, a mapping update will be triggered?

Reply:
@martijnvg it turns out that the `SystemIndexManager` would never have updated the mappings, because without a value for `_metadata.version`, the manager would always think that the version was the latest (because `Version.fromString(null)` returns `Version.CURRENT`). I've pushed a fix for that case.

Reply:
Thanks for fixing that!